site/zenodo_rdm/legacy/services.py

```diff
@@ -84,15 +96,60 @@ class LegacyRecordService(RDMRecordService):
     def create(self, identity, data, uow=None, expand=False):
         """Create a draft and prereserve the DOI."""
         res = super().create(identity, data, uow=uow, expand=False)
-        res = self.pids.create(
-            identity=identity,
-            id_=res.id,
-            scheme="doi",
-            expand=True,
-            uow=uow,
-        )
+        pids = data.get("pids", {})
+
+        # If pids is provided, it is created automatically.
+        # TODO confirm this is correct
+        # TODO what happens with other providers / schemes? They are allowed in RDM
+        if not pids:
+            res = self.pids.create(
+                identity=identity,
+                id_=res.id,
+                scheme="doi",
+                expand=True,
+                uow=uow,
+            )
+
+        communities = data.get("communities", [])
+        p = self.add_communities(communities, identity, uow, res._record)
+        print("------")
+        print(p)
+
+        # TODO how
+        # uow.register(IndexRefreshOp(indexer=current_record_communities_service.indexer))
+
         return res
 
+    def add_communities(self, communities, identity, uow, record):
+        """Creates community inclusion requests for a record's communities.
+
+        This bypasses the record communities service checks on community add.
+        """
+        if not communities:
+            return
+
+        processed = []
+        errored = []
+        for community in communities:
```

Comment:
This code was taken from [rdm-records](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/services/communities/service.py#L129). 

The idea was to bypass the [pid resolve](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/services/communities/service.py#LL125C26-L125C26) because it does not exist yet (the draft is not published)

Reply:
I think we, discussed something similar in the chat, but the idea would be to see if we could convert this minimal prototype into a couple of flags or config that could make this into a feature (or at least possible via normal non-private service calls).